trick_source/sim_services/MonteCarloGeneration/mc_variable.cc

```diff
@@ -116,3 +116,48 @@ MonteCarloVariable::generate_command()
 {
   command = "\n" + variable_name + " = " + assignment;
 }
+
+/*****************************************************************************
+get_type_str
+Purpose:(Return the type of this variable as string. Meant to be overridden
+  by derived classes)
+*****************************************************************************/
+std::string MonteCarloVariable::get_type_str() const
+{
+  switch (get_type()) {
+  // Unreachable case in current implementation.
+  // All current variable classes have been given a "type"
+  default:
+    return(std::string("Undefined_type"));
+    break;
+  case MonteCarloVariable::Calculated:
+    return(std::string("Calculated"));
+    break;
+  case MonteCarloVariable::Constant:
+    return(std::string("Constant"));
+    break;
+  case MonteCarloVariable::Execute:
+    return(std::string("Execute"));
+    break;
+  case MonteCarloVariable::Prescribed:
+    return(std::string("Prescribed"));
+    break;
+  case MonteCarloVariable::Random:
+    return(std::string("Random"));
+    break;
+  }
+}
+
+
+/*****************************************************************************
+summarize_variable
+Purpose:(Provide a string summarizing the attributes of this MonteCarloVariable)
+*****************************************************************************/
+std::string MonteCarloVariable::summarize_variable() const
+{
+  std::ostringstream ss;
+  ss << variable_name << std::string(": type=") << get_type_str() << std::string(",");
```

Comment:
Was there a reason you ended the line printout with a comma?  The other summarize_variable functions don't have the comma at the end

Reply:
I think the comma should be removed here, then prepended to all derived classes instead. In it's current state it results in an extra comma at the end if `summarize_variable` isn't called polymorphically:
```
test.x_file_lookup[2]: type=Prescribed, max_skip=0, is_dependent=1, filename=Modified_data/datafile.txt, column_number=1, first_column_number=1
test.x_fixed_value_double: type=Constant,
```
I'll take a look at making that change here soon. 

Thanks @drcurry2! By the way - am I correct to assume the `key=<value>, key=<value>` format I've implemented here will work for your post processing tool use case? It's a slightly different format than your implementation but all the info is still there.

Reply:
The format works, I just need to make minor edits to the veras processing to read the file with your changes. Already testing it out

Reply:
@ddj116 Are you still going to make the changes you mentioned above ^

Reply:
Yes I believe I made those fixes already. Looks like the branch just merged so it's in Trick now. We'll have to update our reference to the Trick submodule before this goes live in ramtares. Prefer to wait for the next release but we can supersede that if needed, ping me if you still need this capability and it hasn't yet entered ramtares production branches.